File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/fetch_process_node_column_links_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class FetchProcessNodeColumnLinksDetails(object):
    """
    The information needed to obtain column transformations done by a process.
    """

    #: A constant which can be used with the direction property of a FetchProcessNodeColumnLinksDetails.
    #: This constant has a value of "UPSTREAM"
    DIRECTION_UPSTREAM = "UPSTREAM"

    #: A constant which can be used with the direction property of a FetchProcessNodeColumnLinksDetails.
    #: This constant has a value of "BOTH"
    DIRECTION_BOTH = "BOTH"

    #: A constant which can be used with the direction property of a FetchProcessNodeColumnLinksDetails.
    #: This constant has a value of "DOWNSTREAM"
    DIRECTION_DOWNSTREAM = "DOWNSTREAM"

    def __init__(self, **kwargs):
        """
        Initializes a new FetchProcessNodeColumnLinksDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param anchor_node_id:
            The value to assign to the anchor_node_id property of this FetchProcessNodeColumnLinksDetails.
        :type anchor_node_id: str

        :param process_node_id:
            The value to assign to the process_node_id property of this FetchProcessNodeColumnLinksDetails.
        :type process_node_id: str

        :param process_depth:
            The value to assign to the process_depth property of this FetchProcessNodeColumnLinksDetails.
        :type process_depth: int

        :param direction:
            The value to assign to the direction property of this FetchProcessNodeColumnLinksDetails.
            Allowed values for this property are: "UPSTREAM", "BOTH", "DOWNSTREAM"
        :type direction: str

        :param filter:
            The value to assign to the filter property of this FetchProcessNodeColumnLinksDetails.
        :type filter: oci.aidataplatform_dp.models.ProcessNodeColumnLinksFilter

        """
        self.swagger_types = {
            'anchor_node_id': 'str',
            'process_node_id': 'str',
            'process_depth': 'int',
            'direction': 'str',
            'filter': 'ProcessNodeColumnLinksFilter'
        }

        self.attribute_map = {
            'anchor_node_id': 'anchorNodeId',
            'process_node_id': 'processNodeId',
            'process_depth': 'processDepth',
            'direction': 'direction',
            'filter': 'filter'
        }

        self._anchor_node_id = None
        self._process_node_id = None
        self._process_depth = None
        self._direction = None
        self._filter = None

    @property
    def anchor_node_id(self):
        """
        **[Required]** Gets the anchor_node_id of this FetchProcessNodeColumnLinksDetails.
        ID of the lineage anchor node from which the traversal context is derived.


        :return: The anchor_node_id of this FetchProcessNodeColumnLinksDetails.
        :rtype: str
        """
        return self._anchor_node_id

    @anchor_node_id.setter
    def anchor_node_id(self, anchor_node_id):
        """
        Sets the anchor_node_id of this FetchProcessNodeColumnLinksDetails.
        ID of the lineage anchor node from which the traversal context is derived.


        :param anchor_node_id: The anchor_node_id of this FetchProcessNodeColumnLinksDetails.
        :type: str
        """
        self._anchor_node_id = anchor_node_id

    @property
    def process_node_id(self):
        """
        **[Required]** Gets the process_node_id of this FetchProcessNodeColumnLinksDetails.
        ID of process node for which column transformation links are requested.


        :return: The process_node_id of this FetchProcessNodeColumnLinksDetails.
        :rtype: str
        """
        return self._process_node_id

    @process_node_id.setter
    def process_node_id(self, process_node_id):
        """
        Sets the process_node_id of this FetchProcessNodeColumnLinksDetails.
        ID of process node for which column transformation links are requested.


        :param process_node_id: The process_node_id of this FetchProcessNodeColumnLinksDetails.
        :type: str
        """
        self._process_node_id = process_node_id

    @property
    def process_depth(self):
        """
        **[Required]** Gets the process_depth of this FetchProcessNodeColumnLinksDetails.
        Depth of the process node relative to the anchor node within the lineage traversal.


        :return: The process_depth of this FetchProcessNodeColumnLinksDetails.
        :rtype: int
        """
        return self._process_depth

    @process_depth.setter
    def process_depth(self, process_depth):
        """
        Sets the process_depth of this FetchProcessNodeColumnLinksDetails.
        Depth of the process node relative to the anchor node within the lineage traversal.


        :param process_depth: The process_depth of this FetchProcessNodeColumnLinksDetails.
        :type: int
        """
        self._process_depth = process_depth

    @property
    def direction(self):
        """
        **[Required]** Gets the direction of this FetchProcessNodeColumnLinksDetails.
        Direction of the process node with respect to Anchor node.

        Allowed values for this property are: "UPSTREAM", "BOTH", "DOWNSTREAM"


        :return: The direction of this FetchProcessNodeColumnLinksDetails.
        :rtype: str
        """
        return self._direction

    @direction.setter
    def direction(self, direction):
        """
        Sets the direction of this FetchProcessNodeColumnLinksDetails.
        Direction of the process node with respect to Anchor node.


        :param direction: The direction of this FetchProcessNodeColumnLinksDetails.
        :type: str
        """
        allowed_values = ["UPSTREAM", "BOTH", "DOWNSTREAM"]
        if not value_allowed_none_or_none_sentinel(direction, allowed_values):
            raise ValueError(
                "Invalid value for `direction`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._direction = direction

    @property
    def filter(self):
        """
        Gets the filter of this FetchProcessNodeColumnLinksDetails.

        :return: The filter of this FetchProcessNodeColumnLinksDetails.
        :rtype: oci.aidataplatform_dp.models.ProcessNodeColumnLinksFilter
        """
        return self._filter

    @filter.setter
    def filter(self, filter):
        """
        Sets the filter of this FetchProcessNodeColumnLinksDetails.

        :param filter: The filter of this FetchProcessNodeColumnLinksDetails.
        :type: oci.aidataplatform_dp.models.ProcessNodeColumnLinksFilter
        """
        self._filter = filter

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/FetchProcessNodeColumnLinksDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * The information needed to obtain column transformations done by a process.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=FetchProcessNodeColumnLinksDetails.Builder.class)

public final class FetchProcessNodeColumnLinksDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"anchorNodeId", "processNodeId", "processDepth", "direction", "filter"})
    public FetchProcessNodeColumnLinksDetails(String anchorNodeId, String processNodeId, Integer processDepth, LineageDirection direction, ProcessNodeColumnLinksFilter filter) {
        super();
        this.anchorNodeId = anchorNodeId;
        this.processNodeId = processNodeId;
        this.processDepth = processDepth;
        this.direction = direction;
        this.filter = filter;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * ID of the lineage anchor node from which the traversal context is derived.
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("anchorNodeId")
private String anchorNodeId;

        /**
         * ID of the lineage anchor node from which the traversal context is derived.
* 
         * @param anchorNodeId the value to set
         * @return this builder
         **/
        

public Builder anchorNodeId(String anchorNodeId) {
    this.anchorNodeId = anchorNodeId;
    return this;
}
            /**
     * ID of process node for which column transformation links are requested.
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("processNodeId")
private String processNodeId;

        /**
         * ID of process node for which column transformation links are requested.
* 
         * @param processNodeId the value to set
         * @return this builder
         **/
        

public Builder processNodeId(String processNodeId) {
    this.processNodeId = processNodeId;
    return this;
}
            /**
     * Depth of the process node relative to the anchor node within the lineage traversal.
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("processDepth")
private Integer processDepth;

        /**
         * Depth of the process node relative to the anchor node within the lineage traversal.
* 
         * @param processDepth the value to set
         * @return this builder
         **/
        

public Builder processDepth(Integer processDepth) {
    this.processDepth = processDepth;
    return this;
}
            /**
     * Direction of the process node with respect to Anchor node.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("direction")
private LineageDirection direction;

        /**
         * Direction of the process node with respect to Anchor node.
         * @param direction the value to set
         * @return this builder
         **/
        

public Builder direction(LineageDirection direction) {
    this.direction = direction;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("filter")
private ProcessNodeColumnLinksFilter filter;



public Builder filter(ProcessNodeColumnLinksFilter filter) {
    this.filter = filter;
    return this;
}


        public FetchProcessNodeColumnLinksDetails build() {
            FetchProcessNodeColumnLinksDetails model = new FetchProcessNodeColumnLinksDetails(this.anchorNodeId
                , this.processNodeId
                , this.processDepth
                , this.direction
                , this.filter);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(FetchProcessNodeColumnLinksDetails model) {
                this.anchorNodeId(model.getAnchorNodeId());
    this.processNodeId(model.getProcessNodeId());
    this.processDepth(model.getProcessDepth());
    this.direction(model.getDirection());
    this.filter(model.getFilter());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * ID of the lineage anchor node from which the traversal context is derived.
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("anchorNodeId")
    private final String anchorNodeId;

        /**
     * ID of the lineage anchor node from which the traversal context is derived.
* 
     * @return the value
     **/
    
    public String getAnchorNodeId() {
        return anchorNodeId;
    }


        /**
     * ID of process node for which column transformation links are requested.
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("processNodeId")
    private final String processNodeId;

        /**
     * ID of process node for which column transformation links are requested.
* 
     * @return the value
     **/
    
    public String getProcessNodeId() {
        return processNodeId;
    }


        /**
     * Depth of the process node relative to the anchor node within the lineage traversal.
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("processDepth")
    private final Integer processDepth;

        /**
     * Depth of the process node relative to the anchor node within the lineage traversal.
* 
     * @return the value
     **/
    
    public Integer getProcessDepth() {
        return processDepth;
    }

    
        /**
     * Direction of the process node with respect to Anchor node.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("direction")
    private final LineageDirection direction;

        /**
     * Direction of the process node with respect to Anchor node.
     * @return the value
     **/
    
    public LineageDirection getDirection() {
        return direction;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("filter")
    private final ProcessNodeColumnLinksFilter filter;

    
    public ProcessNodeColumnLinksFilter getFilter() {
        return filter;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("FetchProcessNodeColumnLinksDetails(");
        sb.append("anchorNodeId=").append(String.valueOf(this.anchorNodeId));
        sb.append(", processNodeId=").append(String.valueOf(this.processNodeId));
        sb.append(", processDepth=").append(String.valueOf(this.processDepth));
        sb.append(", direction=").append(String.valueOf(this.direction));
        sb.append(", filter=").append(String.valueOf(this.filter));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof FetchProcessNodeColumnLinksDetails)) {
            return false;
        }

        FetchProcessNodeColumnLinksDetails other = (FetchProcessNodeColumnLinksDetails) o;
        return java.util.Objects.equals(this.anchorNodeId, other.anchorNodeId) &&
            java.util.Objects.equals(this.processNodeId, other.processNodeId) &&
            java.util.Objects.equals(this.processDepth, other.processDepth) &&
            java.util.Objects.equals(this.direction, other.direction) &&
            java.util.Objects.equals(this.filter, other.filter);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.anchorNodeId == null ? 43 : this.anchorNodeId.hashCode());
        result = (result * PRIME) + (this.processNodeId == null ? 43 : this.processNodeId.hashCode());
        result = (result * PRIME) + (this.processDepth == null ? 43 : this.processDepth.hashCode());
        result = (result * PRIME) + (this.direction == null ? 43 : this.direction.hashCode());
        result = (result * PRIME) + (this.filter == null ? 43 : this.filter.hashCode());
        return result;
    }


}
